osquery/tables/system/windows/bam.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+constexpr auto kBamRegPath =
+    "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Services\\bam\\%%\\%%";
+
+// Get last execution time
+auto last_execute_time(std::string& assist_data) {
+  std::string last_run_string = assist_data.substr(0, 16);
+
+  // swap endianess
+  std::reverse(last_run_string.begin(), last_run_string.end());
+
+  char temp;
+  for (std::size_t i = 0; i < last_run_string.length(); i += 2) {
+    temp = last_run_string[i];
+    last_run_string[i] = last_run_string[i + 1];

```

Comment:
I'm concerned this might go out of bounds, consider a `last_run_string` of length 1.

Reply:
I added a check to make sure `last_run_string` is of length 16